build/test.js

```diff
@@ -36,6 +36,10 @@ function runFrontendUnitTests(singleRun, doneFn) {
   };
 
   let server = new karma.Server(localConfig, function(failCount) {
+    if (failCount > 0) {
+      // Kill backend server and cluster, if running.
+      gulp.start('kill-all');
```

Comment:
Looks good for me but one thing is missing. There is only one place where on error we kill backend and cluster. This check is only done in function runProtractorTests. So I will extend it for this code: 
```
import sequence from 'run-sequence';

gulp.task('check:create-cluster', function () {
  runSequence('local-up-cluster', ['check'],
    function () {
      gulp.start('kill-all');
  });
});

```
@bryk @cheld  what do you think  ?